cpp/src/strings/convert/convert_urls.cu

```diff
@@ -328,82 +397,50 @@ std::unique_ptr<column> url_decode(
   size_type strings_count = strings.size();
   if (strings_count == 0) return make_empty_column(data_type{type_id::STRING});
 
+  constexpr int num_warps_per_threadblock = 4;
+  constexpr int threadblock_size          = num_warps_per_threadblock * cudf::detail::warp_size;
+  constexpr int char_block_size           = 256;
+  const int num_threadblocks =
+    std::min(65536, cudf::util::div_rounding_up_unsafe(strings_count, num_warps_per_threadblock));
+
   auto offset_count = strings_count + 1;
-  auto d_offsets    = strings.offsets().data<int32_t>() + strings.offset();
+  auto d_in_offsets = strings.offsets().data<offset_type>() + strings.offset();
```

Comment:
Create a `column_device_view` of the input strings column instead.
```
auto d_strings = column_device_view::create(strings, stream);
```

Reply:
Thanks for the suggestion. Using `column_device_view` indeed makes the code much cleaner.

Since `strings_column_view` uses private inheritance, I have to use `strings.parent()` to get the `column_view` version:
```c++
auto const d_strings = column_device_view::create(strings.parent(), stream);
```
This assumes that [strings_column_view::parent](https://github.com/rapidsai/cudf/blob/359be0a24f702926c276d70c1e3f7f533ab63551/cpp/src/strings/strings_column_view.cu#L37) implementation will always return a `column_view` version of itself, which is [not documented](https://github.com/rapidsai/cudf/blob/be25a30ca20f3135f341c51b36cb075b376d5def/cpp/include/cudf/strings/strings_column_view.hpp#L55-L58). Let me know if there's a better way to handle this.

Reply:
This is correct.